src/SixLabors.Core/Primitives/Rectangle.cs

```diff
@@ -452,19 +448,10 @@ public override string ToString()
         }
 
         /// <inheritdoc/>
-        public override bool Equals(object obj) => obj is Rectangle && this.Equals((Rectangle)obj);
+        public override bool Equals(object obj) => obj is Rectangle other && this.Equals(other);
 
         /// <inheritdoc/>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool Equals(Rectangle other) => this.X == other.X && this.Y == other.Y && this.Width == other.Width && this.Height == other.Height;
```

Comment:
One things I'd like to do with the library is really clean up operators. We do it wrong all over the place.

`==` should always call `IEquatable<T>.Equals(T other)` so this statement should be to remove the potential overhead and standardize. (It's probably inlined but I don't want to take the risk).

``` c#
public bool Equals(Rectangle other) => this.X.Equals(other.X) && this.Y.Equals(other.Y) && this.Width.Equals(other.Width) && this.Height.Equals(other.Height);
```